modules/blockbuilder/blockbuilder.go

```diff
@@ -27,14 +28,15 @@ import (
 const (
 	blockBuilderServiceName = "block-builder"
 	ConsumerGroup           = "block-builder"
+	pollTimeout             = 2 * time.Second
 )
 
 var (
 	metricPartitionLag = promauto.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: "tempo",
 		Subsystem: "block_builder",
-		Name:      "partition_lag",
-		Help:      "Lag of a partition.",
+		Name:      "partition_lag_s",
```

Comment:
```suggestion
		Name:      "partition_lag_seconds",
```

https://prometheus.io/docs/practices/naming/#metric-names

Reply:
IMO the number of pending records is also relevant. I think we should keep the original metric and add one for time.

Reply:
👍 re-added, and it is polled in a separate go routine.